cmd/argo-watcher/state/postgres_state.go

```diff
@@ -2,81 +2,90 @@ package state
 
 import (
 	"database/sql"
-	"encoding/json"
 	"fmt"
 	"time"
 
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+	"gorm.io/datatypes"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+
 	"github.com/avast/retry-go/v4"
-	"github.com/golang-migrate/migrate/v4"
 	_ "github.com/lib/pq"
-	"github.com/rs/zerolog/log"
 
-	"github.com/golang-migrate/migrate/v4/database/postgres"
 	_ "github.com/golang-migrate/migrate/v4/source/file"
 
 	"github.com/shini4i/argo-watcher/cmd/argo-watcher/config"
+	"github.com/shini4i/argo-watcher/cmd/argo-watcher/state/state_models"
 	"github.com/shini4i/argo-watcher/internal/models"
 )
 
 type PostgresState struct {
-	db *sql.DB
+	orm *gorm.DB
 }
 
 // Connect establishes a connection to the PostgreSQL database using the provided server configuration.
-// It takes a pointer to a config.ServerConfig parameter and initializes the database connection.
-// The method also performs database migrations using the specified migrations path.
-// If any errors occur during connection establishment or migrations, they are logged and may cause a panic.
-func (state *PostgresState) Connect(serverConfig *config.ServerConfig) {
-	c := fmt.Sprintf("host=%s port=%s user=%s password=%s dbname=%s sslmode=disable", serverConfig.DbHost, serverConfig.DbPort, serverConfig.DbUser, serverConfig.DbPassword, serverConfig.DbName)
+func (state *PostgresState) Connect(serverConfig *config.ServerConfig) error {
+	dsnTemplate := "host=%s port=%s user=%s password=%s dbname=%s sslmode=disable TimeZone=UTC"
+	dsn := fmt.Sprintf(dsnTemplate, serverConfig.DbHost, serverConfig.DbPort, serverConfig.DbUser, serverConfig.DbPassword, serverConfig.DbName)
+
+	// create connection
+	ormConfig := &gorm.Config{}
+	// we can leave logger enabled only for text format
+	if serverConfig.LogFormat != config.LOG_FORMAT_TEXT {
+		// disable logging until we implement zerolog logger for ORM
+		ormConfig.Logger = logger.Default.LogMode(logger.Silent)
+	} else {
+		// output all the SQL queries
+		ormConfig.Logger = logger.Default.LogMode(logger.Info)
+	}
 
-	db, err := sql.Open("postgres", c)
+	// create ORM driver
+	orm, err := gorm.Open(postgres.Open(dsn), ormConfig)
 	if err != nil {
-		panic(err)
+		return err
 	}
 
-	migrationsPath := fmt.Sprintf("file://%s", serverConfig.DbMigrationsPath)
-
-	driver, _ := postgres.WithInstance(db, &postgres.Config{})
-	migrations, _ := migrate.NewWithDatabaseInstance(
-		migrationsPath,
-		"postgres", driver)
-
-	log.Debug().Msg("Running database migrations...")
+	// save orm object
+	state.orm = orm
 
-	switch err = migrations.Up(); err {
-	case migrate.ErrNoChange, nil:
-		log.Debug().Msg("Database schema is up to date.")
-	default:
-		panic(err)
+	// run migrations
+	// note: this doesn't delete existing columns. only adds new ones
```

Comment:
Do we need to do something with the existing setups that were using the previous migrations approach?

Reply:
I created the model [here](https://github.com/shini4i/argo-watcher/pull/166/files#diff-33fa90ab160518ee7316c74f87fd1ee36a874b1e1dc9e71d7cdc4e463dfcf2ceR12-R22) in a way that would use the same columns with similar or the same types. This means that we don't really need to change anything with the existing setup, the auto-migration feature would check that the columns are the same and do nothing.

If we add new columns, those will be added automatically with auto-migrate.